crypto/merklearray/merkle.go

```diff
@@ -0,0 +1,133 @@
+package merklearray
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/crypto"
+)
+
+type Tree struct {
+	// Level 0 is the leaves.
+	levels []layer
+}
+
+func (tree *Tree) topLayer() layer {
+	return tree.levels[len(tree.levels)-1]
+}
+
+func Build(array Array) (*Tree, error) {
+	tree := &Tree{}
+
+	var leaves layer
+	arraylen := array.Length()
+	for i := uint64(0); i < arraylen; i++ {
+		data, err := array.Get(i)
+		if err != nil {
+			return nil, err
+		}
+
+		leaves = append(leaves, crypto.HashObj(data))
```

Comment:
If you're going to store transaction ids in the tree, maybe we can avoid hashing the leafs ? I know that from order of magnitude it would not make a difference, but we might be able to squeeze some more performance out by doing so. In case it has any security reasons, then scrap that.

Reply:
We can just make the data be the transaction in that case.